src/kolibri_android/android_utils.py

```diff
@@ -933,6 +943,94 @@ def setup_analytics():
+
+    __javainterfaces__ = ["android/content/DialogInterface$OnClickListener"]
+    __javacontext__ = "app"
+
+    def __init__(self, activity, package_name):
+        super().__init__()
+        self.activity = activity
+        self.package_name = package_name
+
+    @java_method("(Landroid/content/DialogInterface;I)V")
+    def onClick(self, dialog, which):
+        logger.debug(f"Starting webview update activity for {self.package_name}")
+
+        # Try both the market:// and https://play.google.com/ URIs.
+        uri = Uri.parse(f"market://details?id={self.package_name}")
+        intent = Intent(Intent.ACTION_VIEW, uri)
+        try:
+            self.activity.startActivity(intent)
+        except JavaException as err:
+            if err.classname != "android.content.ActivityNotFoundException":
+                raise
+
+            uri = Uri.parse(
+                f"https://play.google.com/store/apps/details?id={self.package_name}"
+            )
+            intent.setData(uri)
+            try:
+                self.activity.startActivity(intent)
+            except JavaException as err:
+                if err.classname != "android.content.ActivityNotFoundException":
+                    raise
+
+                logger.warning(
+                    f"No activity found to update webview package {self.package_name}"
+                )
+
+
+def check_webview_version():
+    """Check if the current webview meets requirements"""
+    # getCurrentWebViewPackage is only available since API 26.
+    if SDK_INT < 26:
+        logger.warning(f"Cannot get webview package on SDK {SDK_INT}")
+        return
+
+    pkg = WebView.getCurrentWebViewPackage()
+    if pkg is None:
+        logger.warning("Could not determine current webview package")
+        return
+
+    pkg_name = str(pkg.packageName)
+    pkg_version = str(pkg.versionName)
+    logger.info(f'Webview package: {pkg_name} "{pkg_version}"')
+
+    min_major_version = WEBVIEW_MIN_MAJOR_VERSION.get(pkg_name)
+    if min_major_version is None:
+        logger.warning(f"Cannot check webview version for {pkg_name}")
+        return
+
+    try:
+        major_version = int(pkg_version.split(".", 1)[0])
+    except ValueError:
+        logger.warning(f'Could not parse webview major version from "{pkg_version}"')
+        return
+
+    logger.debug(f"{pkg_name} major version: {major_version}")
+    if major_version >= min_major_version:
+        return
+
+    # Create a dialog to instruct the user to update the webview.
+    logger.debug(f"Initiating webview package {pkg_name} update")
+    activity = get_activity()
+    builder = AlertDialogBuilder(activity)
+    builder.setMessage(
+        AndroidString(
+            "The webview component does not meet Endless Key requirements. "
+            "Please update it and restart the application."
+        )
+    )
+    builder.setPositiveButton(
+        AndroidString("Update"), WebViewUpdateClickListener(activity, pkg_name)
+    )
```

Comment:
Am I right in thinking that, by default, you can dismiss the dialog by tapping outside it? It's not completely clear from the SDK documentation.

Reply:
I hadn't actually tested, but you're right. You can set the dialog to not cancelable to force it to be acknowledged. Is that what we want? I assume yes, so I'll add that.

Reply:
I think we should start with that since we know the app will not work below a given webview version, so it really is required.

Reply:
Added `.setCancelable(False)` to the builder to achieve this.